File: uiframe/src/main/java/mobi/cangol/mobile/base/FragmentInfo.java
/**
 * Copyright (c) 2013 Cangol
 * <p/>
 * Licensed under the Apache License, Version 2.0 (the "License")
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 * <p/>
 * http://www.apache.org/licenses/LICENSE-2.0
 * <p/>
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package mobi.cangol.mobile.base;

import android.os.Bundle;

public class FragmentInfo {
    final String tag;
    final Class<? extends BaseFragment> clss;
    final Bundle args;

    public FragmentInfo(Class<? extends BaseFragment> _class, String _tag, Bundle _args) {
        tag = _tag;
        clss = _class;
        args = _args;
    }

    public FragmentInfo(BaseFragment fragment) {
        tag = fragment.getTag();
        clss = fragment.getClass();
        args = fragment.getArguments();
    }

    @Override
    public String toString() {
        return "FragmentInfo [tag=" + tag + ", clss=" + clss + ", args=" + args
                + "]";
    }

}

File: uiframe/src/main/kotlin/mobi/cangol/mobile/base/FragmentInfo.kt
package mobi.cangol.mobile.base

import android.os.Bundle

class FragmentInfo {
    val tag: String?
    val clazz: Class<out BaseFragment>
    val args: Bundle?

    constructor(clazz: Class<out BaseFragment>, tag: String, args: Bundle?) {
        this.tag = tag
        this.clazz = clazz
        this.args = args
    }

    constructor(fragment: BaseFragment) {
        tag = fragment.tag
        clazz = fragment.javaClass
        args = fragment.arguments
    }

    override fun toString(): String {
        return "FragmentInfo [tag=$tag, clazz=$clazz, args=$args]"
    }

}
